examples/pipeline-example.yaml

```diff
@@ -42,6 +42,33 @@
       - [batch2.input.training-labels*, train.input.training-labels]
       - [batch2.metadata.optimal_learning_rate, train.parameter.learning_rate]
       - [batch2.output.processed-images*, train.input.training-images]
+- pipeline:
+    name: My deployment pipeline
+    nodes:
+      - name: batch1
+        type: execution
+        step: Batch feature extraction
+      - name: train
+        type: execution
+        step: Train model
+      - name: deploy
+        type: deployment
+        deployment:
+          name: MyDeployment
+          version:
+            name: 2000102.0
```

Comment:
This can't be static, since attempting to create a version with the same name as a pre-existing version would fail, and there's no such concept as updating a deployment version's configuration; they're very much immutable.

If left empty, we could use the same logic we use for suggesting a new version name in the UI (i.e. generate a `YYYYMMDD.X` version).

I do foresee a need for changing that logic to allow for e.g. a prefix – should we allow some placeholders here? Something like `foo-{datestamp}.{counter}`? (@ruksi, ideas?)

```suggestion
```

Reply:
we can make the `name` optional and we always add the `{stamp}.{counter}` either way; as the full name or just as suffix